vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc

```diff
@@ -142,14 +142,19 @@ void Promotion::OnGetAllPromotions(
 
   if (list.size() > 0) {
     for (auto & item : list) {
-      auto it = promotions.find(item->id);
-      if (it != promotions.end()) {
-        if (promotions.at(item->id)->status !=
-            ledger::PromotionStatus::ACTIVE) {
-          item->status = promotions.at(item->id)->status;
-        }
+      if (item->legacy_claimed) {
+        item->status = ledger::PromotionStatus::CLAIMED;
+        ClaimTokens(item->Clone(), [](const ledger::Result _){});
       } else {
-        promotions.insert(std::make_pair(item->id, item->Clone()));
+        auto it = promotions.find(item->id);
+        if (it != promotions.end()) {
+          if (promotions.at(item->id)->status !=
+              ledger::PromotionStatus::ACTIVE) {
+            item->status = promotions.at(item->id)->status;
+          }
+        } else {
+          promotions.insert(std::make_pair(item->id, item->Clone()));
+        }
```

Comment:
You don't need to call `promotions.at(item->id)` again. There is an iterator `it` which points to the item you need. Use `it->second.status` instead.

Reply:
Done.